### ISSUE TITLE
property with setter misidentified as Method

### ISSUE THREAD

**Issue by mlfz** [PyCQA/pylint#870]
I recently upgrade to pylint-1.5.5

``` sh
pylint --version
pylint 1.5.5, 
astroid 1.4.5
Python 2.7.6 (default, Jun 22 2015, 17:58:13) 
[GCC 4.8.2]
```

And now my code is generating a lot of false-positive no-member errors, one class of which can be summarized by the following code:

``` py
# foo.py
class X(object):
    def __init__(self, foo=None):
        self._foo = foo
    @property
    def foo(self):
        return self._foo
    @foo.setter
    def foo(self, val):
        self._foo = val

if __name__ == '__main__':
    print X([]).foo.append
```

pylint generates the following error:

```
$ pylint -E foo.py                                                                                               
************* Module foo
E: 12,10: Method 'foo' has no 'append' member (no-member)
```

The error goes away if I remove the `foo.setter` method, which is not used in the code above.  So for some reason when `.setter` is defined for an `@property`, pylint thinks the property is a method.

**Comment by PCManticore** [comment]
Interesting, will look into it!

**Comment by dargueta-work** [comment]
Also happening on Python 3.5.1:

```
pylint 1.5.5, 
astroid 1.4.5
Python 3.5.1 (default, Jan 28 2016, 14:55:02) 
[GCC 4.2.1 Compatible Apple LLVM 7.0.2 (clang-700.1.81)]
```

**Comment by titusjan** [comment]
Can I inquire what the status is? This bug affects me too. I now have to disable the `no-member` warning in all modules/functions that **use** class X, not only in the module that defines it. So I would say this bug has a pretty high impact.

```
~> pylint --version
pylint 1.6.4, 
astroid 1.4.7
Python 3.5.2 (default, Jul  6 2016, 10:28:05) 
[GCC 4.9.2]

```

**Comment by NotSqrt** [comment]
I looked into it for a while:

When hitting ``@property``, astroid infers that this node is not a BoundMethod, but when hitting ``foo.setter``, it says that ``foo`` is a ``BoundMethod`` now.

If the setter uses a different name, there is no collision.

In ``astroid.bases._is_property``, ``meth.decoratornames()`` returns an empty list because, even if ``meth.decorators.nodes`` is not empty (``[<Attribute(setter) ...>]``), ``decoratornames`` catches an ``InferenceError``.

I hope this helps to narrow it down.

**Comment by PCManticore** [comment]
@titusjan nobody is working currently on it and I have some busy months lately. Feel free to work on a patch.
